### ISSUE TITLE
Zim 0.70 on MacOS Mojave: Cutting and pasting a word causes crash

### ISSUE THREAD

**Issue by reagle** [zim-desktop-wiki/zim-desktop-wiki#590]
```
...
DEBUG: Loading config from: /Users/reagle/.cache/zim/notebook-Users_reagle_joseph_plan_zim/state.conf
DEBUG: Loading config from: <ConfigFile: /Users/reagle/.config/zim/style.conf>
DEBUG: Autosave interval: 15 - use threads: True
INFO: Open page: Misc:Travel Check List (Misc:Travel Check List)
DEBUG: Action: toggle_panes(False)
DEBUG: Action: toggle_panes(True)
DEBUG: Spellcheck language: en_US
DEBUG: requesting name for code "de_DE"
DEBUG: requesting name for code "fr_FR"
DEBUG: requesting name for code "en_GB"
DEBUG: requesting name for code "en_AU"
DEBUG: requesting name for code "en_US"
DEBUG: Accelmap: /Users/reagle/.config/zim/accelmap
DEBUG: Add window: MainWindow
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: Action: copy
DEBUG: Action: paste
DEBUG: Targets available for paste: [Gdk.Atom.intern("NSStringPboardType", False), Gdk.Atom.intern("UTF8_STRING", False)], we want parsetree
DEBUG: Requesting data for ('text/x-zim-parsetree', 0, 1) -- using set_with_data() workaround
Python(73100,0x1165e85c0) malloc: *** error for object 0x4a: pointer being freed was not allocated
Python(73100,0x1165e85c0) malloc: *** set a breakpoint in malloc_error_break to debug
[1]    73100 abort      /usr/local/bin/python3 $HOME/bin/zim-desktop-wiki-master/zim.py --standalone
```

**Comment by reagle** [comment]
Oh, this is a bad one. Any paste, whether from something copied in zim or elsewhere, is causing a crash.

**Comment by reagle** [comment]
Because this is a Python malloc, should this be reported to python or GTK?

**Comment by jaap-karssenberg** [comment]
These are python gtk bindings - would report to gtk project, believe they also maintain the python bindings.

**Comment by reagle** [comment]
https://gitlab.gnome.org/GNOME/gtk/issues/1584

**Comment by ebassi** [comment]
GTK does not maintain the Python bindings—those are part of [pygobject](https://gitlab.gnome.org/GNOME/pygobject), and the bindings are dynamically generated via introspection data. The crash seems to be caused by pygobject trying to deallocate some memory it shouldn't free. This would either hint to a wrong ownership transfer annotation, or to a bug in pygobject itself.

Would it be possible for the Zim developers to figure out where in their code is the crash?

**Comment by reagle** [comment]
@ebassi thanks for the response! @jaap-karssenberg, any suggestions?

The crash is prompted by [this line](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/master/zim/gui/clipboard.py#L583):

 ```
		atoms = sorted(
					filter(lambda a: a.name() in PARSETREE_ACCEPT_TARGET_NAMES, atoms),
					key=lambda a: PARSETREE_ACCEPT_TARGET_NAMES.index(a.name())
				)
```

**Comment by reagle** [comment]
https://gitlab.gnome.org/GNOME/gtk/issues/1584#note_454644

> Tomasz Miąsko @tmiasko · 1 hour ago. Quartz version of gtk_clipboard_wait_for_targets doesn't have transfer container annotation, maybe that is why it tries to deallocate atoms?

**Comment by reagle** [comment]
Fixed in 3.24.7, thanks for your help.
